### PR TITLE
Poetry conversion

### PR DESCRIPTION
**Pull request by achidlow**
testing GitHub actions

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        require_serial: true
+        types_or: [ python, pyi ]
+      - id: flake8
+        name: flake8
+        description: '`flake8` is a command-line utility for enforcing style consistency across Python projects.'
+        entry: poetry run flake8
+        language: system
+        types: [ python ]
+        require_serial: true
+        # TODO: expand this coverage to examples!
```
Comment (robdmoore) [inline .pre-commit-config.yaml:19]: Do we want to do this?

File: CONTRIBUTING.MD [hunk 2]
```diff
@@ -24,7 +24,12 @@ With few exceptions, each pull request should be initiated against the master br
 
 We make a best-effort attempt to adhere to [PEP 8 - Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/)
 
-A linter is run during CI to perform some basic checks
+A linter is run during CI to perform some basic checks.
+
+After running `poetry install` to set up your Python virtual environment for this project,
```
Comment (robdmoore) [inline CONTRIBUTING.MD:29]: I rekon we should have a proper initial setup section, we could copy from the algokit-cli one?

Comment (achidlow) [reply CONTRIBUTING.MD:29]: I think we can expand on this in another PR, but I wouldn't go to the same level as algokit-cli

File: .pre-commit-config.yaml (hunk 1 above)
Comment (achidlow) [reply .pre-commit-config.yaml:19]: Yes, absolutely. Examples should be exemplary.

**Pull request closed without merging**